tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_cell_data.py

```diff
@@ -17,25 +18,21 @@ def test_enable_cell_data_via_settings_app(self):
         settings.launch()
         cell_and_data_settings = settings.open_cell_and_data_settings()
 
-        # verify that a carrier is displayed
-        self.assertTrue(len(cell_and_data_settings.carrier_name) > 0)
+        self.assertNotEqual(cell_and_data_settings.carrier_name, '')
 
         # enable cell data
         self.assertFalse(cell_and_data_settings.is_data_toggle_checked)
         cell_data_prompt = cell_and_data_settings.enable_data()
 
-        # deal with prompt that sometimes appears (on first setting)
-        if cell_data_prompt.is_displayed:
-            # Cell data should not be enabled until we turn it on via the prompt
-            self.assertFalse(cell_and_data_settings.is_data_toggle_checked)
-            self.assertFalse(self.data_layer.get_setting('ril.data.enabled'), "Cell data was enabled before responding to the prompt")
-            cell_data_prompt.turn_on()
+        self.assertTrue(cell_and_data_settings.is_data_toggle_checked)
+        # Even though the data toggle checkbox is already checked,
+        # cell data should not be enabled until we turn it on via the prompt
+        self.assertFalse(self.data_layer.get_setting('ril.data.enabled'),
+            "Cell data was enabled before responding to the prompt")
 
-        # Wait for cell data to be turned on
-        self.wait_for_condition(lambda m: cell_and_data_settings.is_data_toggle_checked)
+        cell_data_prompt.turn_on()
 
         # verify that cell data is now enabled and connected
         self.assertTrue(self.data_layer.is_cell_data_enabled, "Cell data was not enabled via Settings app")
```

Comment:
That's what self.data_layer.is_cell_data_enabled is doing:
http://mxr.mozilla.org/gaia/source/tests/python/gaia-ui-tests/gaiatest/gaia_test.py#281